src/modules/vtol_att_control/standard.cpp

```diff
@@ -72,17 +72,12 @@ void Standard::update_vtol_state()
 
 	float mc_weight = _mc_roll_weight;
 
-	if (_vtol_vehicle_status->vtol_transition_failsafe) {
+	if (_vtol_vehicle_status->fixed_wing_system_failure) {
 		// Failsafe event, engage mc motors immediately
 		_vtol_mode = vtol_mode::MC_MODE;
 		_pusher_throttle = 0.0f;
 		_reverse_output = 0.0f;
 
-		//reset failsafe when FW is no longer requested
-		if (!_attc->is_fixed_wing_requested()) {
-			_vtol_vehicle_status->vtol_transition_failsafe = false;
-		}
-
```

Comment:
I think this deletion has some side-effects on commander's failsafe. IIRC the failsafe mode (as specified by COM_QC_ACT) is applied for as long as the `vtol_fixed_wing_system_failure` flag is set to true. When it gets cleared, the drone resumes the mode it was in pre-QC.

So your deletion of code here would mean that sending a Transition-to-MC command no longer resets the commander failsafe, changing existing behaviour.

I might be horribly outdated, as the commander module has changed significantly since I last observed this (v1.12-based)

Same goes for tiltrotor and tailsitter, obviously.

Reply:
No it seems to stay on the mode that was triggered through the failsafe of COM_QC_ACT when a transition to FW is triggered an the `vtol_fixed_wing_system_failure` flag is thus reset to false.

Reply:
Good point.

In my defence, in v1.13 it's as I described : on a Transition-to-MC the drone resumed the mission on my test. 

So all good to delete. In any case I prefer the drone to stay in its QC_ACT mode also once the flag is cleared.